File: news_server/modules/source/presentation/http/registry.ts
/**
 * Handler 注册表
 * 类似 Sjgz-Backend 的 interfaces/http/registry.go
 */
import { SourceHandler } from './handler/source'

export class Registry {
  Source: SourceHandler

  constructor(sourceHandler: SourceHandler) {
    this.Source = sourceHandler
  }
}

File: modules/source/interfaces/http/registry.go
package http

import (
	sourceapp "nfxnews/modules/source/application/source"
	"nfxnews/modules/source/interfaces/http/handler"
)

type Registry struct {
	Source *handler.SourceHandler
	I18n   *handler.I18nHandler
}

func NewRegistry(svc *sourceapp.Service, langs string) *Registry {
	return &Registry{Source: handler.NewSourceHandler(svc), I18n: handler.NewI18nHandler(langs)}
}
